mitodl/teachersportal
# Added product API

#### What's this PR do?

Adds a read-only endpoint at `/api/v1/products/` to list all products in oscar database.
#### Where should the reviewer start?

`portal/views.py`
#### How should this be manually tested?

Add a product to the database:
- Start teacher's portal locally and go to http://localhost:8075/oscar/dashboard/catalogue/
- Under 'Create new product of type' choose 'Course' and click the plus next to it
- Put anything you want for title. UPC **must** begin with `Course_` but otherwise can be anything
- On the left click 'Categories' and then select 'Course' which should be the only category available
- Click 'Save and continue editing'
- On the left click 'Variants', then click 'Add variant'
- Fill in whatever title you want. The UPC **must** start with `Module_` but otherwise can be anything
- On the left click 'Stock and Pricing'. Partner should be edX (the only partner which should be available). SKU _must_ be exactly the same as the value you put in the UPC text field (the one that began with `Module_`). Currency should be `$`. Enter a number for "Price (excl tax)" (make sure it's not "Cost price" or something else). Click Save.
- Go to `/api/v1/products/` and verify that you see two products, the course and then the module. It must be at the end of the list since this list should be sorted by creation order.
#### Any background context you want to provide?

This doesn't include a detail page as it doesn't seem necessary right now.
#### What are the relevant tickets?

Fixes #117 
#### What gif best describes this PR or how it makes you feel?

![](https://media4.giphy.com/media/3o85xo8f9wVYcwzZcY/200.gif)
